app/models/devolved_administration_availability.rb

```diff
@@ -4,4 +4,27 @@ class DevolvedAdministrationAvailability
   embedded_in :local_transaction_edition
   field :type, type: String, default: "local_authority_service"
   field :alternative_url, type: String
+
+  validates :type, inclusion: { in: %w[local_authority_service devolved_administration_service unavailable], allow_blank: true }
+  validate :alternative_url_presence
+  validate :alternative_url_format
+
+  def alternative_url_presence
+    if devolved_administration_service_selected? && alternative_url.blank?
+      errors.add(:alternative_url, "must be provided")
+    end
+  end
+
+  def alternative_url_format
+    if devolved_administration_service_selected? && alternative_url.present?
+      uri = URI.parse(alternative_url)
+      unless uri.is_a?(URI::HTTP) || uri.is_a?(URI::HTTPS)
```

Comment:
This looks a neat way to check it's the right type of URL 👍  - I think we'll only want users to enter a HTTPS URL

Reply:
Thanks, I updated.